src/routes/store/index.svelte

```diff
@@ -1,18 +1,89 @@
 <script context="module">
+  import * as api from '@/utils/api.js';
+  const productLimit = 24;
+
   export async function preload() {
-    const res = await this.fetch('/store.json');
-    return {items: await res.json()};
+    const { pages, data } = await api.get(`/products?limit=${productLimit}`);
+    const colors = await api.get('/colors');
+    return { products: data, colors, pages };
   }
 </script>
 
 <script>
+  import { stores } from '@sapper/app';
   import Layout from '@/layouts/default.svelte';
   import Tagline from '@/containers/store/tagline.svelte';
-  import StoreCard from '@/components/store/card.svelte';
+  import Balance from '@/components/store/balance.svelte';
   import Filters from '@/containers/store/filters.svelte';
-  import Ordering from '@/containers/store/ordering.svelte';
+  import StoreCard from '@/components/store/card.svelte';
+  import Pagination from '@/components/common/pagination.svelte';
+
+  const orderOptions = [
+    { orderBy: 'addition_time', order: 'asc' },
+    { orderBy: 'addition_time', order: 'desc' },
+    { orderBy: 'price', order: 'asc' },
+    { orderBy: 'price', order: 'desc' },
+  ];
+  const orderLabels = [
+    'newest first',
+    'newest last',
+    'cheapest first',
+    'cheapest last',
+  ];
+
+  const { session } = stores();
+
+  export let products;
+  export let pages;
+  export let colors;
+
+  let balance;
+  let order = orderOptions[0];
+  let orderLabel = orderLabels[0];
+  let currentPage = 1;
+
+  let filterElement;
+
+  try {
+    balance = $session.user.balance;
```

Comment:
I bet you don't ever have `session.user` because it should be sent by sapper middleware from *src/server.js* just like it is explained here. 
Then imagine you have installed **express-session** and seeded it to sapper. The problem is how to save the session in the first place cause `user` object will still be empty. One solutions is to save it when you have authenticated via client (in `onMount` function), and then use it for future usage without forgetting to delete session after log out.

Reply:
I don't even know why you review this) I've said in the description that I didn't do session interaction yet, this is merely a placeholder

Reply:
Then why add a useless line? PR should be fully working without any errors. You can make it smaller (f.e. just making auth work or displaying product card using preloading), but not a really wide PR that doesn't work

Reply:
It's fully working without errors. This if is merely a reminder to get back to. That's for your first two sentences, the last one i didn't really understand